aptos-move/vm-genesis/src/lib.rs

```diff
@@ -1063,195 +1167,404 @@ pub fn test_genesis_module_publishing() {
 pub fn test_mainnet_end_to_end() {
+    const TOTAL_SUPPLY: u64 = 100_000_000_000 * APTOS_COINS_BASE_WITH_DECIMALS;
+    const PBO_DELEGATOR_STAKE: u64 = 9_000_000 * APTOS_COINS_BASE_WITH_DECIMALS; // 9 mil
+
     use aptos_types::{
         account_address,
         on_chain_config::ValidatorSet,
         state_store::state_key::StateKey,
         write_set::{TransactionWrite, WriteSet},
     };
 
-    let balance = 10_000_000 * APTOS_COINS_BASE_WITH_DECIMALS;
-    let non_validator_balance = 10 * APTOS_COINS_BASE_WITH_DECIMALS;
+    // 21 Staking accounts - Total 210 million $Supra
+    let pbo_balance = 10_000_000 * APTOS_COINS_BASE_WITH_DECIMALS; // 10 million each
+
+    // 7 Operator balances - Total 70 $Supra
+    let operator_balance = 10 * APTOS_COINS_BASE_WITH_DECIMALS; // 10 Supra each
+
+    // 9 employees - Total 270 million $Supra
+    let employee_balance = 30_000_000 * APTOS_COINS_BASE_WITH_DECIMALS; // 30 million each
+
+    // The rest go to Supra Foundation - 99939999870 $Supra
+    // Ninety-Nine Billion, Nine Hundred Thirty-Nine Million, nine hundred ninety-nine thousand, eight hundred seventy.
+    let supra_foundation_balance =
+        TOTAL_SUPPLY - (21 * pbo_balance) - (7 * operator_balance) - (9 * employee_balance);
 
     // currently just test that all functions have the right interface
-    let account44 = AccountAddress::from_hex_literal("0x44").unwrap();
-    let account45 = AccountAddress::from_hex_literal("0x45").unwrap();
-    let account46 = AccountAddress::from_hex_literal("0x46").unwrap();
-    let account47 = AccountAddress::from_hex_literal("0x47").unwrap();
-    let account48 = AccountAddress::from_hex_literal("0x48").unwrap();
-    let account49 = AccountAddress::from_hex_literal("0x49").unwrap();
-    let operator0 = AccountAddress::from_hex_literal("0x100").unwrap();
-    let operator1 = AccountAddress::from_hex_literal("0x101").unwrap();
-    let operator2 = AccountAddress::from_hex_literal("0x102").unwrap();
-    let operator3 = AccountAddress::from_hex_literal("0x103").unwrap();
-    let operator4 = AccountAddress::from_hex_literal("0x104").unwrap();
-    let operator5 = AccountAddress::from_hex_literal("0x105").unwrap();
+    let supra_foundation = AccountAddress::from_hex_literal("0x777").unwrap();
+
+    let test_validators = TestValidator::new_test_set(Some(7), None);
+
+    let operator_addrs0 = test_validators[0].data.operator_address;
+    let operator_addrs1 = test_validators[1].data.operator_address;
+    let operator_addrs2 = test_validators[2].data.operator_address;
+    let operator_addrs3 = test_validators[3].data.operator_address;
+    let operator_addrs4 = test_validators[4].data.operator_address;
+    let operator_addrs5 = test_validators[5].data.operator_address;
+    let operator_addrs6 = test_validators[6].data.operator_address;
+
+    // PBO winner
+    let pbo_account01 = AccountAddress::from_hex_literal("0x01").unwrap();
```

Comment:
Since `0x01` is the address of `supra_framework` I am not sure if this test would pass or whether it would create some problem. I think first 10 or 16 are system accounts and should not be used for testing addresses since the side effects of doing so are not well understood.

Reply:
This test was supposed to be rewritten from scratch again and is in WIP. I've `#[ignore]`-ed this test due to time constraints. Once the Release Candidate tag is published for upcoming rounds, I will raise another cleanup PR to get this test passing again.

Reply:
This debt is now tracked in https://github.com/Entropy-Foundation/aptos-core/issues/46.